url1cbaffb197_foxking0416_AlgorithmTraining_tgz-pJ8-algorithm_leetcode_MainLeetJ8/src/datastructure/avltree/AVLTree.java

```diff
@@ -0,0 +1,223 @@
+package datastructure.avltree;
+import java.util.Stack;
+
+import datastructure.binarytree.*;
+
+public class AVLTree extends BinaryTree {
+	
+
+	
+	@Override
+ 	protected void insertNode(TreeNode currentNode, TreeNode newNode){
+		if(newNode.val < currentNode.val){
+			if(currentNode.getLeftLeafNode() == null){
+				currentNode.setLeftLeafNode(newNode);
+			}
+			else {
+				insertNode(currentNode.getLeftLeafNode(), newNode);
+			}
+		}
+		else{
+			if(currentNode.getRightLeafNode() == null){
+				currentNode.setRightLeafNode(newNode);
+			}
+			else {
+				insertNode(currentNode.getRightLeafNode(), newNode);
+			}
+		}
+		
+		checkBalance(currentNode);
+	}
+	
+	
+	@Override
+	public void remove(TreeNode removeNode){
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 20. By comparison, 99% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.